src/api/index.js

```diff
@@ -301,8 +301,8 @@ export function getAllCredentials(namespace) {
   return get(uri);
 }
 
-export function getCredential(id, namespace) {
-  const uri = getKubeAPI('secrets', { name: id, namespace });
+export function getCredential(name, namespace) {
+  const uri = getKubeAPI('secrets', name, namespace);
```

Comment:
```suggestion
  const uri = getKubeAPI('secrets', { name, namespace });
```

I'm not sure how this is working since `getKubeAPI` expects an object containing the name and namespace as its second param...

The test should be update to ensure the values are being passed correctly to the API, currently fetchMock is just matching 'secrets' in the URL.

Reply:
yeah so I thought it would be done the way you've suggested but passing the variables in like that gives you an encoded url back that doesn't lead anywhere - only works with passing the variables in how I've done it seemingly

Reply:
Also I'm not sure I understand what you mean about changing the test? I just copied how the other functions in `index.js` have been tested

Reply:
ok so the reason this was working is that the `name` parameter being passed to `getCredential` was actually an object containing both the name and namespace

the function should be written like this:

```diff
-export function getAllCredentials(namespace) {
-  const uri = getKubeAPI('secrets', namespace);
-  return get(uri);
-}
-
-export function getCredential(name, namespace) {
-  const uri = getKubeAPI('secrets', name, namespace);
+export function getCredential({ name, namespace }) {
+  const uri = getKubeAPI('secrets', { name, namespace });
   return get(uri);
 }
```

removed `getAllCredentials` as it's not used anywhere, `/src/api/index.test.js` contains a test that can be deleted, and `/src/actions/secrets.test.js` contains a reference to this function that can also be removed without affecting the test (it wasn't doing anything anyway...)

as for the test for `getCredential`, it didn't catch this error so obviously could do with being updated to properly check the  correct values are being passed as expected. We should also review the other tests to ensure we don't have similar gaps but that's beyond the scope of this PR.

Reply:
thanks for explaining this to me, I'll have a go at updating it now